src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -224,6 +124,16 @@ private OnnxNodeInfo[] GetOutputsInfo()
                     _modelManager.GetOutputShapesDict(_modelName, _ignoredVersion));
         }
 
+        public string[] GetInputNames()
+        {
+            return _inputNames.ToArray();
+        }
+
+        public string[] GetOutputNames()
+        {
+            return _outputNames.ToArray();
+        }
```

Comment:
We don't need these new methods. Let just rename _inputNames to InputNames and make it readonly property. And List<> should be fine if you use Contains()

Reply:
Done...made public. We need to use IndexOf, since we need the actual index (if it contains the item)

---
In reply to: [234305812](https://github.com/dotnet/machinelearning/pull/1586#discussion_r234305812) [](ancestors = 234305812)